backend/pkg/controllers/main.controller.go

```diff
@@ -681,6 +680,69 @@ func (c *MainController) RefreshTokenHandler(ctx *gin.Context) {
 	})
 }
 
+func (c *MainController) AuthenticateAgent(ctx *gin.Context) {
+	var user models.User
+	userId, err := getUserIdFromToken(ctx)
+	if err != nil {
+		ctx.JSON(http.StatusUnauthorized, gin.H{"error": err.Error()})
+		return
+	}
+
+	result := c.usersCollection.FindOne(ctx, bson.M{"userId": userId})
+	err = result.Decode(&user)
+	if err != nil {
+		fmt.Printf("Error: %s", err)
+		ctx.JSON(400, gin.H{"error": err.Error()})
+		return
+	}
+	if user.AgentBearerToken != "" {
+		ctx.JSON(http.StatusOK, gin.H{
+			"message": "Success",
+			"token":   user.AgentBearerToken,
```

Comment:
Doesn't the agent need the refresh token?

Reply:
nope agent token exp is not verified it technically never expires